pkg/controller/jobframework/reconciler.go

```diff
@@ -317,9 +317,7 @@ func (r *JobReconciler) ReconcileGenericJob(ctx context.Context, req ctrl.Reques
 		if workload.HasQuotaReservation(wl) {
 			if !job.IsActive() {
 				log.V(6).Info("The job is no longer active, clear the workloads admission")
-				workload.UnsetQuotaReservationWithCondition(wl, "Pending", evCond.Message)
-				_ = workload.SyncAdmittedCondition(wl)
-				err := workload.ApplyAdmissionStatus(ctx, r.client, wl, true)
+				err := unsetQuotaReservation(wl, "Pending", evCond.Message, ctx, r, true)
```

Comment:
I don't think this `unsetQuotaReservation` is needed.

Reply:
Ohh it's a refactor, I included `unsetQuotaReservation` in this PR https://github.com/kubernetes-sigs/kueue/pull/1252/files/76eb96f94bbfe43e0491ae7f98024fcaf5cfd24e#diff-552ab657667de70900b502e2617bcb405fbf4203ff08b2a35d7c87f2d637a75bR872

Reply:
let's just revert if it's not necessary for implementing this feature.